FinniversKit/Sources/Components/NMPInfobox/NMPInfoboxView.swift

```diff
@@ -0,0 +1,201 @@
+            self.informationType = informationType
+            self.title = title
+            self.detail = detail
+            self.linkInteraction = linkInteraction
+            self.primaryButtonInteraction = primaryButtonInteraction
+            self.secondaryButtonInteraction = secondaryButtonInteraction
+        }
+
+        var titleStyle: Label.Style {
+            .bodyStrong
+        }
+
+        var detailStyle: Label.Style {
+            .body
+        }
+    }
+}
+
+#warning("Og så skal der laves en SwiftUI version som rent faktisk bruger den der Infobox ting")
+public struct NMPInfoboxView: View {
+    public let viewModel: ViewModel
+
+    public init(viewModel: ViewModel) {
+        self.viewModel = viewModel
+    }
+
+    public var body: some View {
+        HStack(alignment: .top, spacing: 8) {
+
+            viewModel.informationType.icon
+
+            VStack(alignment: .leading, spacing: 8) {
+                Text(viewModel.title)
+                    .finnFont(viewModel.titleStyle)
+                Text(viewModel.detail)
+                    .finnFont(viewModel.detailStyle)
+
+                if let linkInteraction = viewModel.linkInteraction {
+                    SwiftUI.Button(
+                        action: { linkInteraction.onTapped() },
+                        label: {
+                            Text(linkInteraction.title).underline()
+                        }
+                    )
+                    .buttonStyle(InlineFlatStyle())
+                }
+
+                HStack(spacing: 8) {
+                    if let primaryButtonInteraction = viewModel.primaryButtonInteraction {
+                        SwiftUI.Button(
+                            action: { primaryButtonInteraction.onTapped() },
+                            label: {
+                                Text(primaryButtonInteraction.title)
+
+                            }
+                        )
+                        .buttonStyle(DefaultStyle(fullWidth: false))
+                    }
+
+                    if let secondaryButtonInteraction = viewModel.secondaryButtonInteraction {
+                        SwiftUI.Button(
+                            action: { secondaryButtonInteraction.onTapped() },
+                            label: { Text(secondaryButtonInteraction.title) }
+                        )
+                        .buttonStyle(FlatStyle(fullWidth: false))
+                    }
+
+                    Spacer()
+                }
+            }
+
+        }
+        .padding(.spacingM)
+        .overlay( // border
+            RoundedRectangle(cornerRadius: 4)
+                .stroke(viewModel.informationType.borderColor, lineWidth: 1.0)
+        )
+        .overlay( // sidebar
+            Rectangle()
+                .fill(viewModel.informationType.sideboxColor)
+                .frame(width: 4),
+            alignment: .leading
+        )
+        .frame(maxWidth: /*@START_MENU_TOKEN@*/.infinity/*@END_MENU_TOKEN@*/)
```

Comment:
Placeholder markers shouldn't be there.

Reply:
🤔 interesting. Must've been because I just used the default value, it looked OK in Xcode

Reply:
Yes that's correct, the default is always a placeholder with hidden markup so you have to input something. As you can see it will work fine but the code gets messy. 😄